lib/elixir/lib/exception.ex

```diff
@@ -454,44 +454,52 @@ defmodule Exception do
   end
 
   defp format_sup_data({:invalid_intensity, intensity}) do
-    "invalid intensity: " <> inspect(intensity)
+    "invalid max_restarts (intensity): " <> inspect(intensity)
   end
 
   defp format_sup_data({:invalid_period, period}) do
-    "invalid period: " <> inspect(period)
+    "invalid max_seconds (period): " <> inspect(period)
   end
 
-  defp format_sup_data(other), do: inspect(other)
+  defp format_sup_data(other), do: "got: #{inspect other}"
+
+  defp format_sup_spec({:duplicate_child_name, id}) do
+   """
+   more than one child specification has the id: #{inspect id}
+   If using maps as child specifications, make sure the IDs are unique.
+   If using a module or {module, arg}, use Supervisor.child_spec/3.
+   """
```

Comment:
This is going to be extremely confusing. The new `child_spec` functions generated by `use` does not allow customising this. I would imagine using one of those processes will be the most common scenario for arriving at this error, and to be honest, I don't really see how to solve it without going completely outside the generated `child_spec` function.